docs/how_to_use.md

```diff
@@ -0,0 +1,453 @@
+# How To Use
+
+
+Create a test file in the tests folder
+
+```python
+from pytest_splunk_addon.standard_lib.addon_basic import Basic
+class Test_App(Basic):
+    def empty_method():
+        pass
+```
+
+## Test Execution
+
+There are three ways to execute the tests:
+
+**1. Running tests with an external Splunk instance**
+
+ ```python3
+ pip3 install pytest-splunk-addon
+ ```
+
+ Run pytest with the add-on, in an external Splunk deployment
+
+ ```bash
+ pytest --splunk-type=external --splunk-app=<path-to-addon-package --splunk-data-generator=<path to pytest-splunk-addon-data.conf file --splunk-host=<hostname --splunk-port=<splunk-management-port --splunk-user=<username --splunk-password=<password --splunk-hec-token=<splunk_hec_token
+ ```
+
+**2. Running tests with docker splunk**
+
+ ```bash
+ git clone git@github.com:splunk/pytest-splunk-addon.git
+ cd pytest-splunk-addon
+ pip install poetry
+ poetry install
+ ```
+
+ Create a Dockerfile.splunk file
+```
+ ARG SPLUNK_VERSION=latest
+ FROM splunk/splunk:$SPLUNK_VERSION
+ ARG SPLUNK_VERSION=latest
+ ARG SPLUNK_APP_ID=TA_UNKNOWN
+ ARG SPLUNK_APP_PACKAGE=$SPLUNK_APP_PACKAGE
+ RUN echo Splunk VERSION=$SPLUNK_VERSION
+ COPY deps/apps /opt/splunk/etc/apps/
+ COPY $SPLUNK_APP_PACKAGE /opt/splunk/etc/apps/$SPLUNK_APP_ID
+```
+
+
+ Create a Dockerfile.uf file
+ ```
+ ARG SPLUNK_VERSION=latest
+ FROM splunk/universalforwarder:$SPLUNK_VERSION
+ ARG SPLUNK_VERSION=latest
+ ARG SPLUNK_APP_ID=TA_UNKNOWN
+ ARG SPLUNK_APP_PACKAGE=$SPLUNK_APP_PACKAGE
+ COPY $SPLUNK_APP_PACKAGE /opt/splunkforwarder/etc/apps/$SPLUNK_APP_ID
+ ```
+
+<details>
+<summary>Create docker-compose.yml</summary>
+
+```commandline
+version: "3.7"
```

Comment:
Is there a way to source the content of this file from the repository itself? Same for below file.

Reply:
Discussed in Slack - let's leave it as it is right now.